events-api/protobuf/v1/event.proto

```diff
@@ -67,3 +69,35 @@ enum Version {
   UnknownVersion = 0;
   V1 = 1;
 }
+
+// Event specific details
+message EventDetails {
+  // Rebuild event details
+  optional RebuildEventDetails rebuild_details = 1;
+}
+
+// Rebuild event details
+message RebuildEventDetails {
+  // Rebuild status
+  RebuildStatus rebuild_status = 1;
+  // Rebuild source replica uri
+  string source_replica = 2;
+  // Rebuild destination replica uri
+  string destination_replica = 3;
+  // Rebuild error for RebuildFail event
+  string error = 4;
```

Comment:
error should be optional?

Reply:
Yes. Will change it. Was using empty string when there is no error.